jenkins/deduper.py

```diff
@@ -96,35 +132,30 @@ def stop_duplicates(self, build_data):
         """
         lines = []
         for pr, builds in build_data.iteritems():
-            if len(builds) > 1:
-                # Sort builds by timestamp (newest to oldest)
-                sorted_builds = sorted(
-                    builds, key=itemgetter('timestamp'), reverse=True)
-
-                # Most recently started build is the first in the list
-                current_build_id = sorted_builds[0]['number']
-
-                # Any other builds are assumed to be outdated
-                old_build_ids = [b['number'] for b in sorted_builds[1:]]
-
-                lines.append("PR #{}:".format(pr))
-                lines.append("\tNum running builds: {}".format(
-                    len(sorted_builds)))
-                lines.append("\tCurrent build: {}".format(
-                    current_build_id))
-                lines.append("\tOutdated builds: {}".format(
-                    old_build_ids))
-
-                desc = self._aborted_description(current_build_id, pr)
-
-                for b in old_build_ids:
-                    try:
-                        self.job.stop_build(b)
-                        self.job.update_build_desc(b, desc)
-                    except Exception as e:
-                        logger.error(e.message)
-
-        if lines:
+            if self.one_per_author:
+                # Assemble a dict where the key is the author and the
+                # value is a list of that author's builds.
+                pr_builds = {}
+                for build in builds:
+                    author = Build(build).author
+
+                    # If we had already found a build for that author
```

Comment:
Did you consider using `pr_builds = defaultdict(list)`? 
It would turn these three lines into just `pr_builds[author].append(build)`.